src/framework.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * File: framework.js 
+ */
+var _ = require('underscore');
+var util = require('util');
+
+var componentFactory = require('./noflo-component-factory');
+var stateFactory = require('./create-state');
+var promiseOutput = require('../src/promise-output');
+var vniManager = require('./vni-manager');
+
+/**
+ * This module implements the core of the RDF Pipeline Framework.  
+ */
+module.exports = {
+
+    componentFactory: function( nodeDef, fRunUpdater ) { 
```

Comment:
The framework will need access to the whole wrapper -- not only its fRunUpdater.  We should discuss this on a design teleconference, since it is uncharted territory.